autorest/adal/token.go

```diff
@@ -384,6 +402,41 @@ func NewServicePrincipalTokenFromUsernamePassword(oauthConfig OAuthConfig, clien
 	)
 }
 
+// NewServicePrincipalTokenFromAuthorizationCode creates a ServicePrincipalToken from the
+func NewServicePrincipalTokenFromAuthorizationCode(oauthConfig OAuthConfig, clientID string, clientSecret string, authorizationCode string, redirectURI string, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
```

Comment:
How is the authorization code obtained?

Reply:
It's returned by the credential provider ( e.g. AAD ).

Simple flow is kind of like this:
1. User calls `/auth` endpoint of web app.
2. Web app redirects to AAD and among others includes  the `redirectURI` ( which must match with the `redirectURI` configured on the AAD app that is used).
3. User auths to AAD and gets redirected to the `redirectURI` with the auth code in the request.

Reply:
We will need to add a sample for this.

Reply:
Azure-Samples/azure-sdk-for-go-samples#13